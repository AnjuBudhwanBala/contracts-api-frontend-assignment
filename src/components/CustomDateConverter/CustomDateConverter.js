import moment from 'moment';

export const CustomDateConverter = ({ date }) => {
  const newDate = moment(new Date(date)).format('DD MMM YYYY');
  return newDate;
};

export default CustomDateConverter;
