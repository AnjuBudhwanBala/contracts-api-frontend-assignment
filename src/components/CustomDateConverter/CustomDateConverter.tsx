import React from 'react';
import moment from 'moment';

interface Props {
  date: string;
}

export const CustomDateConverter: React.FC<Props> = props => {
  const { date } = props;

  const newDate = moment(new Date(date)).format('DD MMM YYYY');
  return <>{newDate}</>;
};

export default CustomDateConverter;
