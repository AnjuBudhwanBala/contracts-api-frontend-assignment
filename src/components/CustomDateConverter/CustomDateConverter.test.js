import React from 'react';
import { shallow } from 'enzyme';
import { CustomDateConverter } from './CustomDateConverter';

describe('CustomDateConverter Component', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      date: new Date()
    };

    wrapper = shallow(<CustomDateConverter date={mockProps} />);
  });

  it('should render CustomDateConverter component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
