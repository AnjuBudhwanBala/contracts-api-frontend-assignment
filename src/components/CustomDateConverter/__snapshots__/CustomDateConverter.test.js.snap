// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomDateConverter Component should render CustomDateConverter component 1`] = `ShallowWrapper {}`;
